packages/graph-explorer/src/components/Tabular/controls/ExportControl/ExternalExportControl.tsx

```diff
@@ -49,67 +48,66 @@ export function ExternalExportControl<T extends Record<string, unknown>>({
 
 function ExportOptionsModal<T extends Record<string, unknown>>({
   instance,
-  omittedColumnsIds,
-}: ExportControlProps<T>) {
+  onClose,
+}: ExportControlProps<T> & { onClose: () => void }) {
   const { rows, data, page, columns, columnOrder, visibleColumns } = instance;
   const [format, setFormat] = useState("csv");
   const [name, setName] = useState<string>("");
   const [options, setOptions] = useState<Record<string, boolean>>({});
   const [selectedColumns, setSelectedColumns] = useState(
     columnOrder.reduce<Record<string, boolean>>((init, col) => {
-      init[col] = !omittedColumnsIds?.includes(col) && visibleColumns[col];
+      init[col] = visibleColumns[col];
       return init;
     }, {})
   );
 
   const onExport = useCallback(() => {
-    let currentDataSource: readonly T[] | Row<T>[] = data;
+    // Filter down to only the columns that are selected
+    const columnsToExport = Object.entries(selectedColumns)
+      .filter(([, isSelected]) => isSelected)
+      .map(([id]) => columns.find(c => c.instance.id === id))
+      .filter(c => c != null);
 
-    // Rows are filtered data
-    if (options["include-filters"]) {
-      currentDataSource = rows;
-    }
+    // Map the data from the rows, if needed
+    const dataToExport = (() => {
+      // Rows are filtered data
+      if (options["include-filters"]) {
+        return rows.map(r => r.original);
+      }
 
-    // Page contains only the visible page
-    if (options["only-page"]) {
-      currentDataSource = page;
-    }
+      // Page contains only the visible page
+      if (options["only-page"]) {
+        return page.map(r => r.original);
+      }
+
+      // Unfiltered data
+      return data;
```

Comment:
Is there any need to map the unfiltered data back to `row.original`? Or are they implicitly the correct format?

Reply:
So, this setup is a bit confusing because it deals with the way `react-table` formats the data.

The variables `rows` and `page` come from `react-table` and are abstractions over the data. The original data can be accessed through the property `original`.

However, `data` is the input for `react-table`. It is the unfiltered, non-paged, data that react-table then wraps around. Hence, no need for mapping in this case.